src/lib/pipeline.ml

```diff
@@ -214,33 +225,49 @@ module Full (Bfx: Extended_edsl.Semantics) = struct
       List.map vcfs ~f:(fun (name, s, v) -> name, s, Bfx.filter_to_region v bed)
 
 
-  let qc fqs =
-    Bfx.concat fqs |> Bfx.fastqc
-
-  let rna_bam ~parameters ~reference_build fqs =
-    Bfx.list_map fqs
-      ~f:(Bfx.lambda (fun fq ->
-          Bfx.star ~configuration:star_config ~reference_build fq))
+  let qc fqs = Bfx.concat fqs |> Bfx.fastqc
+
+  (* Makes a list of samples (which are themselves fastqs or BAMs) into one (or
+     two, if paired-end) FASTQs. *)
+  let concat_samples samples =
+    Bfx.list_map
+      ~f:(Bfx.lambda (fun f -> Bfx.concat f))
+      (Bfx.list samples)
+
+  let rna_bam ~parameters ~reference_build samples =
+    let sample_to_bam sample =
+      Bfx.list_map sample
+        ~f:(Bfx.lambda (fun fq ->
+            Bfx.star ~configuration:star_config ~reference_build fq))
+      |> Bfx.merge_bams
+      |> Bfx.picard_mark_duplicates
+        ~configuration:(mark_dups_config parameters.Parameters.picard_java_max_heap)
+      |> Bfx.gatk_indel_realigner
```

Comment:
Realigning the RNA jointly doesn't seem unreasonable, though it's also unclear to me that we even want to run indel realignment on RNA at all, not sure if that tool can handle spliced alignment…need to dig up our issue on this where we figured this out.